### PR TITLE
Feat section validation

### PR DESCRIPTION
**Pull request by mixcas**
- Extract the common functionality of the sections into a single component: `ApplySection`
- Extend `ApplySection` on each section. Each section has it's own render function.
- Unify submit/save into a single method.
- Add checkmark in `applySidebar` for validated sections

### CODE REVIEW

File: app/imports/api/methods.js [hunk 1]
```diff
@@ -31,18 +31,11 @@ export const createApplication = new ValidatedMethod({
...
   validate({ step, data, applicationId }) {
     
-    // Steps is an array, so positions start at 0.
-    // But the step numbers for the routs start at 1. So we
-    // substract the offset to the step param received.
-    step--;
-
-    const validationSchema = StepsInfo[step].schema;
-
     return new SimpleSchema({
```
Comment (varffs) [inline app/imports/api/methods.js]: where does the schema come from now?

File: app/imports/api/methods.js [hunk 2]
```diff
@@ -31,18 +31,11 @@ export const createApplication = new ValidatedMethod({
 
 // This method could be named better and maybe live
 // in a separate file.
-export const saveApplicationSection = new ValidatedMethod({
-  name: 'application.section.save',
+export const submitApplicationSection = new ValidatedMethod({
+  name: 'application.section.submit',
 
   validate({ step, data, applicationId }) {
```
Comment (varffs) [inline app/imports/api/methods.js:37]: what does this do now?

File: app/imports/components/apply/applySection.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      data,
+    }, (err) => {
+      if (err) {
+        return new Meteor.Error(err);
+      }
+    });
+
+    return callback();
+  }
+
```
Comment (varffs) [inline app/imports/components/apply/applySection.jsx:56]: should there be a default render function as a fallback?

File: app/imports/components/apply/applySidebar.jsx [hunk 4]
```diff
@@ -12,11 +12,21 @@ export class ApplySidebar extends Component {
       let applicationSection = ApplicationSections.findOne({ step: (key + 1) });
 
       // todo add boolean for if section is validated and complete
-      // console.log(applicationSection);
+      //console.log(applicationSection);
+      let validated = false;
+
+      if (!_.isUndefined(applicationSection)) {
+
+        if (!_.isUndefined(applicationSection.validated)) {
```
Comment (varffs) [inline app/imports/components/apply/applySidebar.jsx]: could this be a double conditional rather than 2?

**Comment by varffs** [comment]
All seems to work well and code makes sense. However when you hit the submit button if it validates it doesnt take you to the next step. Doesn't have to be in this PR but should happen

**Comment by varffs** [comment]
Also we need something to deal with section where that are no inputs just text [or do we have a checkbox that has to be selected to validate that section?]

File: app/imports/api/methods.js (hunk 2 above)
Comment (mixcas) [inline app/imports/api/methods.js:37]: This only validates types. For `data` it only checks it an object.

Validation against schema happens inside `run()`.

File: app/imports/components/apply/applySection.jsx (hunk 3 above)
Comment (varffs) [inline app/imports/components/apply/applySection.jsx:56]: @mixcas answer this then it's ready to merge

Comment (mixcas) [inline app/imports/components/apply/applySection.jsx:56]: I dont think is needed. I cant think of any case where it would help or something.

Comment (varffs) [inline app/imports/components/apply/applySection.jsx:56]: just good practice? but ok

Comment (mixcas) [inline app/imports/components/apply/applySection.jsx:56]: what would be the good practice?

Comment (varffs) [inline app/imports/components/apply/applySection.jsx:56]: having a default value set for something essential to the functioning of the class

Comment (mixcas) [inline app/imports/components/apply/applySection.jsx:56]: Nah, cuz this class is meant to be extended. If it's not its useless

Comment (varffs) [inline app/imports/components/apply/applySection.jsx:56]: Yes but if extended without a render function everything breaks. Also for code reference perhaps an example of the render function would be good in the base class?

Comment (mixcas) [inline app/imports/components/apply/applySection.jsx:56]: Ill think about it, i think i might have an idea

**Pull request merged**